src/core/google-auth.js

```diff
@@ -72,3 +74,251 @@ XMPPSocialProvider.prototype.login = function(loginOpts, continuation) {
   }
   this.connect(continuation);
 };
+
+// Returns Promise<{refreshToken, accessToken, email}>
+XMPPSocialProvider.prototype.getCredentialsFromStorage_ = function() {
+  return new Promise(function(fulfill, reject) {
+    this.loadLastRefreshTokenAndEmail_().then(function(data) {
```

Comment:
Just added error checking for this (loadLastRefreshTokenAndEmail_ now rejects if no data is found in storage) - this shouldn't happen in the uProxy case however because uProxy will always set remember=true on the first login (causing the refresh token + email to be written to storage)